### ISSUE TITLE
Stuttering in YTUHD 2160p but perfect in vanilla YouTube

### ISSUE THREAD

**Issue by UprisingPulse** [PoomSmart/YTUHD#6]
Hi, I got iPhone 13 Pro Max, in the vanilla YouTube app from appstore I have the option to play 2160p resolution smoothly, however when I sideloaded YouTube with YTUHD, when I choose 2160p it stutter and buffer.

**Comment by PoomSmart** [comment]
That makes sense. Your sideloaded version of YTUHD probably doesn't include VP9 entitlement spoofing.

**Comment by UprisingPulse** [comment]
So you mean the problem is with the YouTube ipa itself or the tweak?

**Comment by PoomSmart** [comment]
The issue is that you didn't compile YTUHD for use with your IPA with VP9 entitlement spoofing code uncommented.

**Comment by UprisingPulse** [comment]
Affirmative. And how can I do that?

**Comment by PoomSmart** [comment]
@UprisingPulse Assuming you can compile YTUHD from source, line 6-7 and line 60-65 are to be uncommented from [Common.x](https://github.com/PoomSmart/YTUHD/blob/master/Common.x).

**Comment by UprisingPulse** [comment]
I'm getting this error

`> Making all for tweak YTUHD…
==> Compiling Common.x (arm64)…
Common.x:61:10: error: implicit declaration of function 'CFStringEqual' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
     if (CFStringEqual(entitlement, CFSTR("com.apple.coremedia.allow-alternate-video-decoder-selection"))) {
         ^
Common.x:61:10: note: did you mean 'CFStringPad'?
/Users/user/theos/sdks/iPhoneOS14.4.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFString.h:666:6: note: 'CFStringPad' declared here
void CFStringPad(CFMutableStringRef theString, CFStringRef padString, CFIndex length, CFIndex indexIntoPad);
     ^
1 error generated.
make[3]: *** [/Users/user/YTUHD/ytuhd/.theos/obj/debug/arm64/Common.x.a713ed27.o] Error 1
make[2]: *** [/Users/user/YTUHD/ytuhd/.theos/obj/debug/arm64/YTUHD.dylib] Error 2
make[1]: *** [internal-library-all_] Error 2
make: *** [YTUHD.all.tweak.variables] Error 2`

**Comment by PoomSmart** [comment]
https://github.com/PoomSmart/PSHeader/blob/master/Misc.h#L21

**Comment by UprisingPulse** [comment]
> Making all for tweak YTUHD…
==> Linking tweak YTUHD (arm64)…
Undefined symbols for architecture arm64:
  "_SecTaskCopyValueForEntitlement", referenced from:
      __logosLocalCtor_c16daf74 in Common.x.a713ed27.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [/Users/user/ytuhd/.theos/obj/debug/arm64/YTUHD.dylib] Error 1
make[2]: *** [/Users/user/ytuhd/.theos/obj/debug/arm64/YTUHD.dylib] Error 2
make[1]: *** [internal-library-all_] Error 2
make: *** [YTUHD.all.tweak.variables] Error 2
user@MacBook-Pro-User ytuhd % -v 
zsh: command not found: -v

**Comment by PoomSmart** [comment]
This is weird. Do you use the latest version of THEOS?

But then I got you a sideloaded version of YTUHD:

[com.ps.ytuhd_1.2.5_iphoneos-arm.deb.zip](https://github.com/PoomSmart/YTUHD/files/7588161/com.ps.ytuhd_1.2.5_iphoneos-arm.deb.zip)

**Comment by UprisingPulse** [comment]
Yes I have the latest build, I will reinstall it and try to work it out
Thank you very much, it’s working perfectly

**Comment by UprisingPulse** [comment]
Sorry to bother u again, but it’s not working anymore, even with ur deb and I complied you source and injected it, it stopped working, it stutter like before on 2160p resolution

**Comment by PoomSmart** [comment]
Reopen for later investigation.

**Comment by UprisingPulse** [comment]
I injected the deb you sent above to youtube 16.54.5, you can see the stutter after sideloading and enabling vp9 and 2160p res.
[https://github.com/UprisingPulse/test2/releases/download/ww/YouTube_16.45.4+ytuhd.ipa](url)

**Comment by PoomSmart** [comment]
@UprisingPulse With the latest version of everything, does it work? Can you also post the screenshot of "stats for nerds" for the video you are testing with?

**Comment by UprisingPulse** [comment]
Just tried, no it still doesn't work, I have the latest YouTube with the latest everything "tweaks".

This is the screenshots:
![L4z7l4C](https://user-images.githubusercontent.com/33938264/146399337-4cc4aa48-e985-42c6-834a-c6d4a9579c04.jpg)
![ez65rKB](https://user-images.githubusercontent.com/33938264/146399346-f07ae6a7-0ee3-4a09-8626-8f032d8f9c95.jpg)

**Comment by PoomSmart** [comment]
@UprisingPulse So from the screenshot in the Mystery field the value doesn't have `HV`. Are you sure that your dylib has entitlement spoofing code? You can compile the tweak with `make SIDELOADED=1`

**Comment by UprisingPulse** [comment]
`user@MacBook-Pro-User ~ % cd /Users/user/YTUHD 
user@MacBook-Pro-User YTUHD % make SIDELOADED=1
==> Notice: Build may be slow as Theos isn’t using all available CPU cores on this computer. Consider upgrading GNU Make: https://github.com/theos/theos/wiki/Parallel-Building
> Making all for tweak YTUHD…
==> Linking tweak YTUHD (arm64)…
Undefined symbols for architecture arm64:
  "_SecTaskCopyValueForEntitlement", referenced from:
      __logosLocalCtor_199eaf8d in Common.x.f67ceb60.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [/Users/user/YTUHD/.theos/obj/debug/arm64/YTUHD.dylib] Error 1
make[2]: *** [/Users/user/YTUHD/.theos/obj/debug/arm64/YTUHD.dylib] Error 2
make[1]: *** [internal-library-all_] Error 2
make: *** [YTUHD.all.tweak.variables] Error 2
user@MacBook-Pro-User YTUHD % 
`
Got this error again.

**Comment by PoomSmart** [comment]
Copy Xcode’s SDK into $THEOS/sdks and then add PrivateFrameworks folder to it.

**Comment by UprisingPulse** [comment]
Copied Xcode SDK (15.0) to sdks and added PrivateFrameworks from the latest SDK 14.4 from theos repo.
Still getting the same error.

**Comment by PoomSmart** [comment]
@UprisingPulse Could you try adding a framework in Makefile like this:

```
$(TWEAK_NAME)_FRAMEWORKS = Security
```

**Comment by PoomSmart** [comment]
Also, this dylib you may use: https://github.com/qnblackcat/uYouPlus/files/7738656/YTUHD.dylib.zip

**Comment by UprisingPulse** [comment]
Tried my own complied deb and yours dylib

I got this:
![0753F6E8-B470-4E73-B896-BA266287F9F6](https://user-images.githubusercontent.com/33938264/146634864-e55ce610-e171-4f96-88eb-29495931ccb4.jpeg)

still it doesn’t show HV but not stuttering for now, need to test it longer.

**Comment by UprisingPulse** [comment]
Tried my own complied deb and yours dylib

I got this:
![DE8B0456-B368-4E43-86DA-875645CD60E7](https://user-images.githubusercontent.com/33938264/146634924-a2ebe571-7893-4ba5-9cd0-4604ba869ed4.jpeg)

still it doesn’t show HV but not stuttering for now, need to test it longer.

**Comment by UprisingPulse** [comment]
It still stuttering ;(
![2D85568E-0952-4D67-A738-F3F01EA96F56](https://user-images.githubusercontent.com/33938264/146648279-cc562164-ed6e-441b-af39-8a03e5790dc6.png)

**Comment by UprisingPulse** [comment]
It still stuttering

![F3FEC6E5-6F0F-4363-B433-4CC66F91273C](https://user-images.githubusercontent.com/33938264/146648407-88d2d252-12c6-4ad9-ad39-2f787a9d4b13.png)

**Comment by qnblackcat** [comment]
For me, it only happens when playing at 2160p. Not with 1440p. YTUHD 1.2.6

**Comment by PoomSmart** [comment]
Do open a new issue if you think it stills persist with YTUHD 2.0.0
